### DISCUSSION TITLE
Is it possible to index native transfers?

### DISCUSSION THREAD

**Discussion by kevinfaveri** [enviodev/hyperindex#361] [Q&A]
It is perfectly clear how to index ERC stuff, but does envio also supports indexing native transfer of assets so I can get a notification once a transfer in / out of the native gas asset happens in a specific address?

**Comment by PaulRBerg** [comment]
Can we get an answer here, please? @DZakh @JonoPrest @JasoonS 

This feature would be super helpful as we charge our users in ETH

**Reply by DZakh** [reply]
No built-in support, but it is possible to make it work using the Block Handlers feature. Here's the indexer on traces I recently built https://github.com/enviodev/all-contracts-indexer

**Reply by DZakh** [reply]
Note: HyperSync traces are only supported by Eth Mainnet and Gnosis, so you'd need to use RPC for other chains.

**Reply by PaulRBerg** [reply]
I see. I suppose this is because [internal txs are not obtainable from a JSON-RPC server](https://ethereum.stackexchange.com/q/3417/24693), right?

**Reply by JonoPrest** [reply]
They are obtainable via `trace_block` method which is very slow and only available on expensive archive nodes.

**Comment by JonoPrest** [comment]
It won't be super simple until EIP 7708 lands:
https://ethereum-magicians.org/t/eip-7708-eth-transfers-emit-a-log/20034/1

In order to detect native token transfers currently you need look at every transaction and internal transaction (trace) that has a value > 0 and index the from/to fields for balances.

The main blocker for this is its not so easy to support a full chains traces so we only have for gnosis and eth right now.
